### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by werdlerk**
Summary of changes:
- Added Gemfile.lock for the Jekyll static site generator
- Updated layout/styling a bit
  - Changed "Powered by Zilverline" to the Zilverline logo. 
  - Moved link to source code to GitHub icon at top right.
  - Enabled site wide search
  - Enabled copy code button for all highlighted code blocks.
  - Added sitemap.xml for better SEO
  - Created new section "Handbook". These contain pages previously from "Reference" which didn't reference an exact technical item to explain. The Reference section now contains only pages that explain technical references.
  - Moved guide for integrating Sequent with Rails to new section "Handbook"
  - Made all pages use the class `wide` so the pages are wider to display more content. 
  On pages that have a TOC, this class is removed so the space on the right can be used for the TOC.

- Updated home page
- Updated all introduction pages
- Updated Rails & Sequent page

### CODE REVIEW

**Comment by werdlerk** [comment]
Question:
- At the page Modelling the domain, an AggregateRoot is created to ensure uniqueness for Author email addresses, called `Usernames`. This is a Singleton. Why is this named plural? Other AggregateRoots are singular and this being a Singleton, I would expect it also being named `Username` instead.

**Comment by lvonk** [comment]
> * At the page Modelling the domain, an AggregateRoot is created to ensure uniqueness for Author email addresses, called `Usernames`. This is a Singleton. Why is this named plural? Other AggregateRoots are singular and this being a Singleton, I would expect it also being named `Username` instead.

Since it is a collection of usernames.

> On top of that, as it's defined for the purpose of maintaining usernames (uniqueness of email addresses), would a name like UsernameMaintainer be more adequate?

It depends your domain experts. In this fictional case it was me and I chose `Usernames` as most appropriate name.

`UsernameMaintainer` sounds really technical to me, so I think `UsernameList` or `Usernames` is better.

**Comment by werdlerk** [comment]
We need to add a graphic overview of all the steps that the blog application and sequent does. The graphic from the [Lars and Bob’s presentation at GOTO Amsterdam](https://gotocon.com/dl/goto-amsterdam-2013/slides/BobForma_and_LarsVonk_EventSourcingInProductionSystems.pdf) (page 34) is probably a good starting point.

File: docs/docs/concepts/command.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Commands form the API of your domain. Like [Events](event.html), they are simple
 Commands have descriptive names describing the intent of what you are trying to achieve, for example `CreateUser` or `SendInvoice`.
 Commands inherit from `Sequent::Command`. Additionally
 you can add [Validations](validations.html) to commands to ensure correctness. Sequent uses
-[ActiveModel::Validations](http://api.rubyonrails.org/classes/ActiveModel/Validations.html)
+[ActiveModel::Validations](http://api.rubyonrails.org/classes/ActiveModel/Validations.html){:target="_blank"}
```
Comment (lvonk) [inline docs/docs/concepts/command.md:9]: I don't like opening a new page by default, people can use cmd-click for this and leave it up to them.

Comment (werdlerk) [reply docs/docs/concepts/command.md:9]: I'm sorry, I don't agree. I think it's nice to have links on the same website open in the same window, while external links open in a new tab. This way users stay at the website and don't accidentally click away never to come back.

Comment (lvonk) [reply docs/docs/concepts/command.md:9]: Sure, you can do that by cmd click. Especially on mobile it is annoying when a new tab/window opens. But as this highly depends on the user we should keep it optional. By enforcing a new window, we enforce everyone to use it like this.

**Review by lvonk — COMMENTED** [review]
Nice work ❤️ .

File: docs/_data/navigation.yml [hunk 2]
```diff
@@ -1,62 +1,66 @@
...
         url: /docs/modelling-the-domain.html
-      - title: "3 Building a web application"
+      - title: 3. Building a web application
         url: /docs/building-a-web-application.html
-      - title: "4 Finishing the web application"
+      - title: 4. Finishing the web application
         url: /docs/finishing-the-web-application.html
-      - title: "5 Rails & Sequent"
+
+  - title: Handbook
```
[the remark this answers is not in the document]
Comment (werdlerk) [reply docs/_data/navigation.yml]: If you agree, I would like  to name it "Guides" as there is already an overwhelming amount of "Sequent" mentions 😄

File: docs/docs/building-a-web-application.md [hunk 3]
```diff
@@ -16,13 +19,13 @@ gem 'sinatra'
...
 ```
 
-And then run `bundle install`. We will set up Sinatra to run as a [modular application](https://github.com/sinatra/sinatra#serving-a-modular-application).
-
-To make use of Sinatra, we need to create / modify the following files:
+And then run `bundle install`. We will set up Sinatra to run as
+a [modular application](https://github.com/sinatra/sinatra#serving-a-modular-application){:target="_blank"}.
 
-Create `./app/web.rb`:
+Create `app/web.rb`:
```
[the remark this answers is not in the document]
Comment (werdlerk) [reply docs/docs/building-a-web-application.md:28]: I've tried to keep it consistent everywhere. Technically `./folder` and `folder` are exactly the same as they both work from the current folder. 
If we want to make it clear that it's the folder from the root, we should rename it to `/folder` everywhere.

File: docs/docs/building-a-web-application.md [hunk 4]
```diff
@@ -371,105 +374,76 @@ end
...
+I, [..]  INFO -- : groups: 4096
+I, [..]  INFO -- : Start replaying events
+I, [..]  INFO -- : Done migrate_online for version 2
+I, [..]  INFO -- : Start migrate_offline for version 2
+I, [..]  INFO -- : Number of groups 16
+I, [..]  INFO -- : groups: 16
+I, [..]  INFO -- : Start replaying events
+I, [..]  INFO -- : Migrated to version 2
 ```
+{: .no-copy}
```
[the remark this answers is not in the document]
Comment (werdlerk) [reply docs/docs/building-a-web-application.md:394]: Removes the "copy to clipboard" button in the code block.

**Review by lvonk — APPROVED** [review]

**Pull request merged**